flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/SwitchedSources.java

```diff
@@ -45,4 +47,14 @@ public SimpleVersionedSerializer<SourceSplit> serializerOf(int sourceIndex) {
     public void put(int sourceIndex, Source source) {
         sources.put(sourceIndex, Preconditions.checkNotNull(source));
     }
+
+    public Integer getSourceIndex(int subtaskSourceIndex) {
+        Integer integer = sources.firstKey();
```

Comment:
This can be simplified and I think for readability we can get rid of the extra method:
```
        return previousSubtaskSourceIndex == -1 ? sources.firstKey() : previousSubtaskSourceIndex++
```

Reply:
Note that in the -1 case, the index should be `sources.firstKey()`, not `sources.firstKey() - 1`

Reply:
The index via `subtaskSourceIndex++` , afterwards it is still `sources.firstKey()`.

Reply:
For the refactoring, it's good to me.

Reply:
For the refactor, it's good to me. However I made a small change. `previousSubtaskSourceIndex++` -> `++previousSubtaskSourceIndex`.